File: web-front-reactive/src/main/java/template/modules/admin/command/users/DeleteUserCommand.java
package template.modules.admin.command.users;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;
import template.modules.admin.domain.users.User;
import template.modules.admin.domain.users.UserRepository;

@Service
@AllArgsConstructor
public class DeleteUserCommand {
    private final UserRepository repo;

    public Boolean execute(Long id) {
        Mono<User> existingUser = findById(id);
        if (existingUser != null) {
            repo.deleteById(id);
            return true;
        } else
            return false;
    }

    private Mono<User> findById(Long id) {
        return repo.findById(id);

    }
}

File: web-front-reactive2/src/main/kotlin/template/modules/admin/command/users/DeleteUserCommand.kt
package template.modules.admin.command.users

import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.springframework.stereotype.Service
import template.modules.admin.domain.users.UserRepository

@Service
class DeleteUserCommand(private val repo: UserRepository) {
    suspend fun execute(id: Long): Boolean {
        val existingUser = findById(id)
        return if (existingUser != null) {
            repo.deleteById(id).awaitSingleOrNull()
            true
        } else
            false
    }

    private suspend fun findById(id: Long) =
        repo.findById(id).awaitSingleOrNull()
}